src/main.c

```diff
@@ -134,13 +134,12 @@ static gboolean check_graphical_sessions(gpointer data) {
   }
   return G_SOURCE_REMOVE;
 }
+
 static void do_startup(GObject *object, gpointer data) {
   notify_init("snapd-desktop-integration");
   client = snapd_client_new();
-  refresh_monitor = sdi_refresh_monitor_new();
-  if (!sdi_refresh_monitor_start(
-          refresh_monitor,
-          g_application_get_dbus_connection(G_APPLICATION(app)), NULL)) {
+  refresh_monitor = sdi_refresh_monitor_new(G_APPLICATION(object));
+  if (!sdi_refresh_monitor_start(refresh_monitor, NULL)) {
```

Comment:
(not caused by this PR). Just noting the GError is set to NULL - it should be caught and written into the g_message below.

Reply:
You are right. Done.